Sming/Components/OtaUpgrade/component.mk

```diff
@@ -0,0 +1,192 @@
+# Future configurability (ATM, there is only signing)
+
+# If enabled, OTA firmware images are digitally signed and the upgrade mechanism will reject
+# images with an invalid signature.
+# You should not disable this feature unless there is absolutely no way your device is accessed
+# through a public network.
```

Comment:
> Coding style. Although it's not mandatory for Components or libraries I think it would be appropriate to
> apply the Sming coding style to OtaStream.

Could the build system be extended such that Components or Libraries may "opt in" for clangformat?

Reply:
I tend to run it manually (`make cs` takes forever in Windows) so have got into the habit of doing Ctrl+Shift+F in Eclipse (or Source/Format from context menu), which picks up the default `.clang-format` file.

Enforcing this via Travis CI would mean updating the main Makefile (make cs target) but it doesn't do any component parsing. I suspect `component-wrapper.mk` would do the work for each Component, which would then need to be invoked via project.mk.

Could add an 'all or nothing' flag, so it just pulls in the resultant set of files from `COMPONENT_SRCDIRS`, `COMPONENT_SRCFILES` and `COMPONENT_INCDIRS`.

Overall it's probably not that trivial, but definitely do-able. If its behaviour is useful and can be easily documented then probably worth doing....

Reply:
https://github.com/SmingHub/Sming/pull/1999#issuecomment-568212558